tpm2/test/src/cipher/aes_gcm_test.c

```diff
@@ -56,6 +63,411 @@ int aes_gcm_add_tests(CU_pSuite suite)
+                                 uint8_t ** result_vec,
+                                 size_t * result_vec_len,
+                                 bool * expect_pass)
+{
+  // create buffer to hold vector data read in from file a line at a time
+  // specify buffer size to handle largest vector component (must include
+  // some extra space for leading and/or trailing characters that get
+  // stripped off
+  char buffer[MAX_TEST_VECTOR_COMPONENT_LENGTH];
+
+  // create stack variables to buffer the components in a single test vector
+  uint8_t  * Key = calloc(MAX_TEST_VECTOR_COMPONENT_LENGTH, 1);
+  size_t Key_len = 0;
+  uint8_t * IV = calloc(MAX_TEST_VECTOR_COMPONENT_LENGTH, 1);
+  size_t IV_len = 0;
+  uint8_t * CT = calloc(MAX_TEST_VECTOR_COMPONENT_LENGTH, 1);
+  size_t CT_len = 0;
+  uint8_t * AAD = calloc(MAX_TEST_VECTOR_COMPONENT_LENGTH, 1);
+  size_t AAD_len = 0;
+  uint8_t * Tag = calloc(MAX_TEST_VECTOR_COMPONENT_LENGTH, 1);
+  size_t Tag_len = 0;
+  uint8_t * PT = calloc(MAX_TEST_VECTOR_COMPONENT_LENGTH, 1);
+  size_t PT_len = 0;
+  bool pass_result = false;
+
+  // create/initialize a flag to report that an applicable kmyth vector found
+  bool test_vector_found = false;
+
+  // create/initialize a counter to track progress (ensure that test vector
+  // components are read and parsed in expected sequence)
+  //     step = 1: looking for a 'Key' vector component
+  //     step = 2: expecting 'IV' vector component
+  //     step = 3: expecting 'CT' vector component
+  //     step = 4: expecting 'AAD' vector component
+  //     step = 5: expecting 'Tag' vector component
+  //     step = 6: expecting 'PT' vector component or 'FAIL'
+  //               check kmyth applicability of test vector
+  // any other parsing sequence values are invalid - failure or unexepected
+  // file data at any step restarts the process (reset to first step)
+  int step = 1;
+
+  // test vector file is read a line at a time until either EOF is reached
+  // or a test vector grouping applicable to kmyth is successfully parsed.
+  while (!feof(fid) && !test_vector_found)
+  {
+    if (fgets(buffer, MAX_TEST_VECTOR_COMPONENT_LENGTH, fid) != NULL)
+    {
+      if ((strncmp(buffer, "Key = ", 6) == 0) &&
+          (step == 1))
+      {
+        // process 'Key' component of this test vector
+        char * key_str = buffer + 6;  // strip preceding 'Key = ' sub-string
+        int key_str_len = strlen(key_str);
+        while ( (key_str_len > 0) &&
+                ((key_str[key_str_len-1] == '\n') ||
+                 (key_str[key_str_len-1] == '\r')) )
+        {
+          key_str[--key_str_len] = '\0';  // strip any trailing '\n' or 'r'
+        }
+        Key_len = key_str_len / 2;  // 2 hex chars map to a byte of key
+        convert_HexString_to_ByteArray((char **) &Key,
+                                       key_str,
+                                       key_str_len);
+        Key_len = key_str_len / 2;  // 2 hex chars map to a byte of key
```

Comment:
I don't see any reason to do this twice (lines 133 and 137), but in case it's important I didn't change it when I was editing the comments.

Reply:
This must have been a copy/paste remnant that I didn't clean up properly. I deleted the redundant assignment.